crates/sage-cli/src/runner.rs

```diff
@@ -251,18 +307,21 @@ impl Runner {
                 if level != 2 && level != 3 {
                     log::warn!("TMT quant level set at {}, is this correct?", level);
                 }
-                sage_core::tmt::quantify(&spectra, isobaric, Tolerance::Ppm(-20.0, 20.0), level)
+                sage_core::tmt::quantify(&msn_spectra, isobaric, Tolerance::Ppm(-20.0, 20.0), level)
             })
             .unwrap_or_default();
-        let ms1 = spectra.into_iter().filter(|s| s.level == 1).collect();
 
         SageResults {
             features,
             quant,
-            ms1,
+            ms1: ms1_spectra,
         }
     }
 
+    fn requires_ms1(&self) -> bool {
```

Comment:
this will presumably get inlined by the compiler, but what's the point of this function?

Reply:
tbh it felt nicer to read when "pushing down" the requirement to read the MS1 data when loading the spectra ('read ms1 if requires_ms1()' felt more intuitive than 'read ms1 if config.parameters.quant.lfq'). which I made an opt-in bc reading the MS1 frames is significantly more expensive than the MS2 in bruker data.